src/Authoring/WinRT.SourceGenerator/MergeReferencedActivationFactoriesGenerator.cs

```diff
@@ -0,0 +1,199 @@
+            });
+
+        // Generate the chaining helper
+        context.RegisterImplementationSourceOutput(assemblyExportsTypeNames, static (context, assemblyExportsTypeNames) =>
+        {
+            if (assemblyExportsTypeNames.IsEmpty)
+            {
+                context.AddSource("ChainedExports.g.cs", """
+                    // <auto-generated/>
+                    #pragma warning disable
+
+                    namespace WinRT
+                    {
+                        using global::System;
+
+                        /// <inheritdoc cref="Module"/>
+                        partial class Module
+                        {
+                            /// <summary>
+                            /// Tries to retrieve the activation factory from all dependent WinRT components.
+                            /// </summary>
+                            /// <param name="fullyQualifiedTypeName">The marshalled fully qualified type name of the activation factory to retrieve.</param>
+                            /// <returns>The pointer to the activation factory that corresponds with the class specified by <paramref name="fullyQualifiedTypeName"/>.</returns>
+                            internal static IntPtr TryGetDependentActivationFactory(ReadOnlySpan<char> fullyQualifiedTypeName)
+                            {
+                                return default;
+                            }
+                        }
+                    }
+                    """);
+            }
+            else
+            {
+                StringBuilder builder = new();
+
+                builder.AppendLine("""
+                    // <auto-generated/>
+                    #pragma warning disable
+
+                    namespace WinRT
+                    {
+                        using global::System;
+
+                        /// <inheritdoc cref="Module"/>
+                        partial class Module
+                        {
+                            /// <summary>
+                            /// Tries to retrieve the activation factory from all dependent WinRT components.
+                            /// </summary>
+                            /// <param name="fullyQualifiedTypeName">The marshalled fully qualified type name of the activation factory to retrieve.</param>
+                            /// <returns>The pointer to the activation factory that corresponds with the class specified by <paramref name="fullyQualifiedTypeName"/>.</returns>
+                            internal static unsafe IntPtr TryGetDependentActivationFactory(ReadOnlySpan<char> fullyQualifiedTypeName)
+                            {                  
+                    """);
+
+                foreach (string assemblyExportsTypeName in assemblyExportsTypeNames)
+                {
+                    builder.AppendLine($$"""
+                                    IntPtr obj = global::{{assemblyExportsTypeName}}.GetActivationFactory(fullyQualifiedTypeName);
```

Comment:
Is repeating this going to hit multiple variable declaration of the same name?

Reply:
Ah, good catch! Fixed 🙂